building-and-packaging/material/packaging_debian_and_spack_exercise.md

```diff
@@ -0,0 +1,99 @@
+
+## Optional Tasks
+
+If you work on any of the optional tasks, please mention this in your pull request.
+
+### CPack
+
+If you work on any of the optional tasks, please mention this in your pull request.
+
+We want to automatize the package creation using a Docker container. Extend your Docker container image such that it automatically creates the Debian image without further user interaction. The Debian image should be copied to the directory `/mnt/package-build-dir` within the running container as `/mnt/package-build-dir` will be mounted at runtime of the container.
+
+Ultimately, on will only run the following (or similar) command
+
+```bash
+docker run --rm -it --name cpack-packaging --mount type=bind,source="$(pwd)",target=/mnt/package-build-dir IMAGENAME
+```
+
+Please state the actual command to use in your pull request.
+
+Running the container like this should trigger the preparation of the two packages. After the container is finished running, the files `PACKAGENAME.deb` and `PACKAGENAME.tar.gz` should be present on the host system but no `build` directory or similar, i.e., make sure the package is not created in the mounted drive.
+
+### Spack
+
+- Add the main branch of the GitHub repository [as version to the Spack recipe](https://spack.readthedocs.io/en/latest/packaging_guide.html#git).
+- Make the current dependencies (`Boost`, `yaml-cpp`) optional by defining suitable options in the `CMakeLists.txt` which are also picked up in the C++ code. Then add corresponding variants in the Spack recipe.
+- Create a Spack package for the Python code that you have packaged with pip. Checkout [Spack's documentation regaring the `PythonPackage`](https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html)  class for recipes if you want to work on this.
+
+## Remarks and Tips
+
+- If you need to specify a homepage, you can use the course homepage `https://simulation-software-engineering.github.io/homepage/`.
+- All parts of the exercise should be ideally done on Docker container based on Ubuntu 20.04.
+- If you are in a container and your libraries in `/usr/local/lib` are not found, you can extend the `LD_LIBRARY_PATH` accordingly.
+- If you need an inspiration on how to write more complicated Spack recipes, you can check packages that are already shipped included with spack (`spack edit PACKAGENAME` or browse).
+- If you have `Boost` or `deal.ii` installed in the Docker container via `apt`, finding pre-installed packages using `spack external find` might fail. In this case you have to define the dependencies manually or run `spack external find` before installing the critical packages.
+- Consider using the Docker image `ajaust/spack-package-building-base` as base for the Spack exercise.
```

Comment:
From dockerhub?

Reply:
Yes from Dockerhub. it is mentioned in the exercise, but not in the hints... I will fix that.